### ISSUE TITLE
RPC eth_listening and eth_mining always returning false

### ISSUE THREAD

**Issue by jorisbontje** [ethereum/go-ethereum#323]
With a mist node that is mining and connected to 2 peers, the RCP methods `eth_listening` and `eth_mining` always return False.

Request:
```
{'params': None, 'jsonrpc': '2.0', 'id': '54eab6cd-b22b-4328-9fa2-ba56f4ddea0a', 'method': 'eth_listening'}
```

Reponse:
```
{u'jsonrpc': u'2.0', u'id': u'54eab6cd-b22b-4328-9fa2-ba56f4ddea0a', u'result': False}
```

expected: result = True

Request:
```
{'params': None, 'jsonrpc': '2.0', 'id': '33a739cf-0f31-47af-b42e-52f5871c0cc0', 'method': 'eth_mining'}
```
Response:
```
{u'jsonrpc': u'2.0', u'id': u'33a739cf-0f31-47af-b42e-52f5871c0cc0', u'result': False}
```

expected: result = True

**Comment by jorisbontje** [comment]
`eth_listening` is now correctly reporting, `eth_mining` is still always false when starting `ethereum -rpc -mine=true`, it works in `mist` though.

**Comment by jorisbontje** [comment]
@obscuren please reopen as `eth_mining` is not correctly fixed when running: `ethereum -rpc=true -mine=true`

**Comment by jorisbontje** [comment]
Fixed by fdf939a6f9b5360d76415c7118969d92af2774f9
